app/main.py

```diff
@@ -39,51 +48,257 @@ def execute(self):
+                    oxp_url=self.dict_shelve['oxp_url'],
+                ).parse_convert_topology()
+            return {"result": topology_mock.topology_mock(),
+                    "status_code": 200}
+        except Exception as err:  # pylint: disable=W0703
+            log.info(err)
+            return {"result": "Validation Error", "status_code": 400}
+
+    def post_sdx_lc(self, event_type):
+        """ return the status from post sdx topology to sdx lc"""
+        post_topology = requests.post(
+                settings.SDX_LC_TOPOLOGY,
+                timeout=10,
+                json=self.sdx_topology)
+        if post_topology.status_code == 200:
+            if event_type != 0:
+                # open the topology shelve
+                with shelve.open("topology_shelve") as open_shelve:
+                    open_shelve['version'] = self.sdx_topology["version"]
+                    open_shelve['timestamp'] = self.sdx_topology["timestamp"]
+                    open_shelve['nodes'] = self.sdx_topology["nodes"]
+                    open_shelve['links'] = self.sdx_topology["links"]
+                    # now, we simply close the shelf file.
+                    open_shelve.close()
+            return {"result": self.sdx_topology,
+                    "status_code": post_topology.status_code}
+        return {"result": post_topology.json(),
+                "status_code": post_topology.status_code}
+
+    def post_sdx_topology(self, event_type=0, event_timestamp=None):
+        """ return the topology following the SDX data model"""
+        # pylint: disable=W0201
+        try:
+            if event_type != 0:
+                topology_update = self.convert_topology(
+                        event_type, event_timestamp)
+                self.sdx_topology = {
+                        "id": topology_update["id"],
+                        "name": topology_update["name"],
+                        "version": topology_update["version"],
+                        "model_version": topology_update["model_version"],
+                        "timestamp": topology_update["timestamp"],
+                        "nodes": topology_update["nodes"],
+                        "links": topology_update["links"],
+                        }
+            else:
+                self.sdx_topology = topology_mock.topology_mock()
+            evaluate_topology = self.validate_sdx_topology()
+            if evaluate_topology["status_code"] == 200:
+                result = self.post_sdx_lc(event_type)
+                return result
+            return {"result": evaluate_topology['result'],
+                    "status_code": evaluate_topology['status_code']}
+        except Exception as err:  # pylint: disable=W0703
+            log.info(err)
+        return {"result": "No SDX Topology loaded", "status_code": 401}
+
     @listen_to("kytos/topology.*")
-    def listen_topology(self, event=None):
+    def listen_event(self, event=None):
```

Comment:
The decorator looks good. The current subscribed events aren't high frequency (since link up also has a 10 sec timer, so even a flap won't overwhelm), so even with the requests will make maybe a single thread will still perform well, it might be worth stress testing this part and confirming in practice the expected behavior.